ai-ml/jupyterhub/addons.tf

```diff
@@ -0,0 +1,244 @@
+  cluster_endpoint  = module.eks.cluster_endpoint
+  cluster_version   = module.eks.cluster_version
+  oidc_provider_arn = module.eks.oidc_provider_arn
+  #---------------------------------------
+  # Amazon EKS Managed Add-ons
+  #---------------------------------------
+  eks_addons = {
+    aws-ebs-csi-driver = {
+      service_account_role_arn = module.ebs_csi_driver_irsa.iam_role_arn
+      preserve                 = true
+      most_recent              = true
+    }
+    coredns = {
+      most_recent = true
+      preserve    = true
+    }
+    kube-proxy = {
+      most_recent = true
+      preserve    = true
+    }
+  }
+  
+  #---------------------------------------
+  # Karpenter Autoscaler for EKS Cluster
+  #---------------------------------------
+  enable_karpenter                  = true
+  karpenter_enable_spot_termination = true
+  karpenter_node = {
+    create_iam_role          = true
+    iam_role_use_name_prefix = false
+    # We are defining role name so that we can add this to aws-auth during EKS Cluster creation
+    iam_role_name = local.karpenter_iam_role_name
+  }
+
+  karpenter = {
+    timeout             = "300"
+    repository_username = data.aws_ecrpublic_authorization_token.token.user_name
+    repository_password = data.aws_ecrpublic_authorization_token.token.password
+  }
+
+}
+
+#---------------------------------------------------------------
+# IRSA for EBS
+#---------------------------------------------------------------
+module "ebs_csi_driver_irsa" {
+  source                = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version               = "~> 5.14"
+  role_name             = format("%s-%s", local.name, "ebs-csi-driver")
+  attach_ebs_csi_policy = true
+  oidc_providers = {
+    main = {
+      provider_arn               = module.eks.oidc_provider_arn
+      namespace_service_accounts = ["kube-system:ebs-csi-controller-sa"]
+    }
+  }
+  tags = local.tags
+}
+
+#---------------------------------------------------------------
+# IRSA for VPC CNI
+#---------------------------------------------------------------
+module "vpc_cni_irsa" {
+  source                = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version               = "~> 5.14"
+  role_name             = format("%s-%s", local.name, "vpc-cni")
+  attach_vpc_cni_policy = true
+  vpc_cni_enable_ipv4   = true
+  oidc_providers = {
+    main = {
+      provider_arn               = module.eks.oidc_provider_arn
+      namespace_service_accounts = ["kube-system:aws-node"]
+    }
+  }
+  tags = local.tags
+}
+#---------------------------------------------------------------
+# VPC CNI Addon should run before the nodes are created
+# Ideally VPC CNI with custom configuration values should be deployed before the nodes are created to use the correct VPC CNI config
+#---------------------------------------------------------------
+resource "aws_eks_addon" "vpc_cni" {
```

Comment:
why is the defined separately and not within the addons module similar to coredns and kubeproxy?

Reply:
updated